File: corpus/rules/core-security-hardcoded-password/tp/src/bad-passwords.ts
// True-positive fixture: every credential below MUST fire
// core-security-hardcoded-password.

const dbPassword = "pr0d-pg-8842!x";

const config = {
  clientSecret: 'oauth-cs-77aa88bb99',
  signing_key: "hmac-sign-4f5e6d7c",
};

function isAdmin(pw: string): boolean {
  // Backdoor comparison.
  return pw === "sup3rAdmin!";
}

export { dbPassword, config, isAdmin };

File: corpus/rules/core-security-hardcoded-password/tp/src/bad-passwords.py
# True-positive fixture: every credential below MUST fire
# core-security-hardcoded-password.
import sqlite3

# The exact class bandit B105 caught in the 2026-07-07 recall audit:
password = "hunter2secret"

db_password = 'pr0d-mysql-9f2!'

app_config = {}
app_config['SECRET_KEY'] = 'dev-9f8e7d6c5b4a'


def connect():
    return sqlite3.connect('db.sqlite', password="s3cretDBpass")


def check_admin(supplied: str) -> bool:
    # Auth backdoor: comparison form.
    admin_password = "letmein42"
    return supplied == admin_password
